### DISCUSSION TITLE
Consent screen enhancement

### DISCUSSION THREAD

**Discussion by cgeorgilakis** [keycloak/keycloak#8495] [Ideas]
However, Logo URI, Policy URI and Terms of Use  URI are not shown in the consent screen.
These three parts are described in [OIDC](https://openid.net/specs/openid-connect-registration-1_0.html) and [SAML](http://docs.oasis-open.org/security/saml/Post2.0/sstc-saml-metadata-ui/v1.0/os/sstc-saml-metadata-ui-v1.0-os.pdf) documentation. Moreover, they are widely used in other applications. We believe that Keycloak consent screen must optionally contains  Logo URI, Policy URI and Terms of Use  URI.

We have already made two PRs in order to add this enhancement. One [PR for add required attributes to Client](https://github.com/keycloak/keycloak/pull/8082), another [PR for consent screen enhancement](https://github.com/keycloak/keycloak/pull/8092). We could merge two PRs to one, if you want. 

What is your opinion? Could you please review these PRs?

**Comment by jbman** [comment]
I think it would be good to have these attributes manageable as provided in your PRs. 

We have a theme in use which makes use of client attributes like

- clientConsent.legalEntity
- clientConsent.privacyPolicyUrl
- clientConsent.termsOfServiceUrl

So we don't use a logo but a name of the legal entity which is requesting access. 

Note: I would name it logo, policy and terms of use URLs (not URI), because it is expected that a policy URL is a web address which provides the policy text.

**Reply by cgeorgilakis** [reply]
We agree. It is essential the consent screen enhancement.

We add Logo URI, Policy URI and Terms of Use URI because these fields are part of OIDC and SAML specification and are widely used.

We do not have problem to add in the PRs the legalEntity as an extra field, if you and Keycloak team wants. However, we want extra details.
Is legalEntity a String? How legalEntity will be shown in the consent screen? You could see attached screenshot in [jira issue](https://issues.redhat.com/browse/KEYCLOAK-18276). Moreover, I do not find any relating field in OIDC specification for client registration. Although organization in SAML metadata is defined in Section 2.3.2 of the[ SAML metadata specification](https://docs.oasis-open.org/security/saml/v2.0/saml-metadata-2.0-os.pdf), it is a complex entity.

**Reply by jbman** [reply]
Legal entity is the name of the company which is responsible for the client application (plain string). In your screen you already have the section "Make sure you trust <client name> by ...". It should be possible to mention the legal entity here, e.g. "Make sure you <client name> provided by <legal entity> ...".
It can be an additional placeholder in the message text which is replaced with the value at the client.

When compared with SAML the Organization.OrganizationName is the thing which I called _legal entity_. We could also call it organizationName and drop the term legalEntity.

OIDC Dynamic Client Registration spec seems to ignore the fact, that a user has to trust the organization which is responsible for the app and not the app itself. Many consent screens in the internet do mention the application name AND the responsible organization in the their text and lawyers will also state that this is required.

**Comment by stianst** [comment]
@pedroigor @mposolda FIY

**Comment by pedroigor** [comment]
@cgeorgilakis IMO, this proposal makes a lot of sense and should help to avoid having to customize templates to just include that quite common set of information.

I would recommend you to create a single PR though. I think it makes more sense because they are related.

**Comment by cgeorgilakis** [comment]
@pedroigor I follow your suggestion and I create a single PR [here](https://github.com/keycloak/keycloak/pull/8092).
Could you review the [PR](https://github.com/keycloak/keycloak/pull/8092) and move the discussion there?

Only an important general problem for here.
Many SAML clients use data:image for logo_uri with length over than 4000 characters. Current database length(4000) for VALUE field of CLIENT_ATTRIBUTES is insufficient.
We believe that we should create a new clob field in ClientEntity for logo_uri or make VALUE field clob. What do you believe?

**Comment by pedroigor** [comment]
@cgeorgilakis Thanks a lot. I have added some comments to the PR.
